src/librustdoc/html/render/print_item.rs

```diff
@@ -1492,61 +1496,134 @@ fn item_constant(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, c: &cle
-                     {}{}<a href=\"#fields\" class=\"anchor\">§</a>\
-                 </h2>\
-                 {}",
-                if s.ctor_kind.is_none() { "Fields" } else { "Tuple Fields" },
-                document_non_exhaustive_header(it),
-                document_non_exhaustive(it)
-            );
-            for (index, (field, ty)) in fields.enumerate() {
+        fn render_struct<'b>(&'b self) -> impl fmt::Display + Captures<'a> + 'b + Captures<'cx> {
+            display_fn(move |f| {
+                let cx = self.cx.borrow();
+                let v = render_struct(
+                    self.it,
+                    Some(&self.s.generics),
+                    self.s.ctor_kind,
+                    &self.s.fields,
+                    "",
+                    true,
+                    *cx,
+                );
+                write!(f, "{v}")
+            })
+        }
+
+        fn document<'b>(&'b self) -> impl fmt::Display + Captures<'a> + 'b + Captures<'cx> {
+            display_fn(move |f| {
+                let mut cx = self.cx.borrow_mut();
+                let v = document(*cx, self.it, None, HeadingOffset::H2);
+                write!(f, "{v}")
+            })
+        }
+
+        fn fields(&self) -> impl Iterator<Item = (&clean::Item, &clean::Type)> {
+            self.s.fields.iter().filter_map(|item| match *item.kind {
+                clean::StructFieldItem(ref ty) => Some((item, ty)),
+                _ => None,
+            })
+        }
+
+        fn should_render_fields(&self) -> bool {
+            matches!(self.s.ctor_kind, None | Some(CtorKind::Fn))
+                && self.fields().peekable().peek().is_some()
+        }
+
+        fn document_non_exhaustive_header(&self) -> &str {
+            document_non_exhaustive_header(self.it)
+        }
+
+        fn document_non_exhaustive(&self) -> impl fmt::Display + 'a {
+            document_non_exhaustive(self.it)
+        }
+
+        fn render_field_in_span<'b>(
+            &'b self,
+            index: &'b usize,
+            field: &'b clean::Item,
+            ty: &'b clean::Type,
+        ) -> impl fmt::Display + Captures<'a> + 'b + Captures<'cx> {
+            display_fn(move |f| {
+                let mut cx = self.cx.borrow_mut();
                 let field_name =
                     field.name.map_or_else(|| index.to_string(), |sym| sym.as_str().to_string());
                 let id = cx.derive_id(format!("{}.{}", ItemType::StructField, field_name));
+                let ty = ty.print(*cx);
                 write!(
-                    w,
+                    f,
                     "<span id=\"{id}\" class=\"{item_type} small-section-header\">\
-                         <a href=\"#{id}\" class=\"anchor field\">§</a>\
-                         <code>{field_name}: {ty}</code>\
-                     </span>",
+                        <a href=\"#{id}\" class=\"anchor field\">§</a>\
```

Comment:
Why is it not part of the template?

Reply:
Tried to put this into the template multiple times, but it raised errors.

Let me try once more and in the case where it still throws errors, I'll share the implementation to see how we can fix it.

Reply:
Finally got it, and found out that some values are HTML strings, causing tests to fail. 

Adding a `safe` filter solves this issue.